lib/vim_caret.ahk

```diff
@@ -0,0 +1,48 @@
+class VimCaret{
+  __New(vim){
+    global VimScriptPath
+    this.Vim := vim
+    this.caretwidths := {"Normal": 10
+                 , "Visual": 10
+                 , "Insert": 1
+                 , "Default": 1}
+  }
+
+  SetCaret(Mode=""){
+    if (this.Vim.Conf["VimChangeCaretWidth"]["val"] == 0){
+      return
+    }
+    width :=
+    if this.Vim.State.IsCurrentVimMode("Vim_Normal"){
+      width := this.caretwidths["Normal"]
+    }else if this.Vim.State.StrIsInCurrentVimMode("Visual"){
+      width := this.caretwidths["Visual"]
+    }else if this.Vim.State.IsCurrentVimMode("Insert"){
+      width := this.caretwidths["Insert"]
+    }else{
+      width := this.caretwidths["Default"]
+    }
+    SetCaretWidth(width)
+  }
+}
+
```

Comment:
Is there any reason for that following functions are outside of the class?
If no, they should be in the class to be hidden, which is better for users who include vim_ahk in their scripts.